src/web-socket-server.ts

```diff
@@ -1,24 +1,61 @@
-import WebSocket from "ws";
+import { createServer } from "http";
+import { Server as IOServer, Socket } from "socket.io";
 import { MessageBus } from "./message-bus";
+import { messageBuilder } from "./message";
+import { generateObject } from "./tests/utils/generateRandomMessage";
+import { test } from "./parameters";
 
-export const webSocketServer = (messageBus: MessageBus, port: number) => {
-  const wss = new WebSocket.Server({ port });
+const AGENT_ID = "0";
 
-  messageBus.subscribe((message) => {
-    wss.clients.forEach((client: WebSocket) => {
-      if (client.readyState === WebSocket.OPEN) {
-        client.send(JSON.stringify(message));
-      }
-    });
+function webSocketServer(
+  messageBus: MessageBus,
+  port: number,
+  agentIds: string[]
+): void {
+  const httpServer = createServer();
+  const io = new IOServer(httpServer, {
+    cors: {
+      origin: "*",
+    },
   });
 
-  wss.on("connection", (ws: WebSocket) => {
-    console.log("WebSocket client connected");
-    ws.on("message", (message: string) => {
-      console.log("Received: %s", message);
+  io.on("connection", (socket: Socket) => {
+    console.log("A user connected");
+    let testStream: any;
+
+    function generateObjectEveryTwoSeconds(): void {
+      console.log(generateObject());
+      socket.emit("message", generateObject());
+      testStream = setTimeout(generateObjectEveryTwoSeconds, 2000);
+    }
+
+    if (test) {
+      generateObjectEveryTwoSeconds();
+    }
```

Comment:
Something about this leaves a bad taste in my mouth. I'd rather not have the core code responsible for generating test data based on a CLI parameter. Rather, it would be better if we could stand up a normal server and then have an external test script pump in fake data if we want to run tests.

Reply:
im ok with this, this was a quick and easy process to help build the FE without a mock data server running.  Will remove